etcdserver/api/etcdhttp/peer_test.go

```diff
@@ -156,3 +186,60 @@ func TestServeMembersGet(t *testing.T) {
 		}
 	}
 }
+
+// TestNewPeerHandlerOnMembersPromotePrefix verifies the request with members promote prefix is routed correctly
+func TestNewPeerHandlerOnMembersPromotePrefix(t *testing.T) {
+	ph := newPeerHandler(zap.NewExample(), &fakeServer{cluster: &fakeCluster{}}, fakeRaftHandler, nil)
+	srv := httptest.NewServer(ph)
+	defer srv.Close()
+
+	tests := []struct {
+		path      string
+		wcode     int
+		checkBody bool
+		wKeyWords string
+	}{
+		{
+			// not a valid path for "POST" method
+			peerMembersPath,
+			http.StatusBadRequest,
+			false,
+			"bad path",
+		},
+		{
+			// does not contain member id in path
+			peerMemberPromotePrefix,
+			http.StatusNotFound,
+			false,
+			"not found in cluster",
+		},
+		{
```

Comment:
Somehow include a test case where the promote succeeds?

Reply:
It is not possible in unit test, as we do not have a real server to promote member. But I will be adding an integration test to send request to follower, which will exercise the promote succeeded path.

Reply:
Integration is fine. Having a configurable `fakeServer` where you can configure it to simulate a successful response might also be a option.